src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageReferenceCommands/ListPackage/ListPackageCommandRunner.cs

```diff
@@ -22,147 +23,142 @@ internal class ListPackageCommandRunner : IListPackageCommandRunner
+            //file and the name of the project
+            Project project = MSBuildAPIUtility.GetProject(projectPath);
+            var projectName = project.GetPropertyValue(ProjectName);
+            ListPackageProjectModel projectModel = listPackageReportModel.CreateProjectReportData(projectPath: projectPath, projectName);
 
-                var assetsPath = project.GetPropertyValue(ProjectAssetsFile);
+            if (!MSBuildAPIUtility.IsPackageReferenceProject(project))
+            {
+                projectModel.AddProjectInformation(problemType: ProblemType.Error,
+                    string.Format(CultureInfo.CurrentCulture, Strings.Error_NotPRProject, projectPath));
+                return;
+            }
 
-                // If the file was not found, print an error message and continue to next project
-                if (!File.Exists(assetsPath))
-                {
-                    Console.Error.WriteLine(string.Format(CultureInfo.CurrentCulture,
-                        Strings.Error_AssetsFileNotFound,
-                        projectPath));
-                    Console.WriteLine();
-                }
-                else
+            var assetsPath = project.GetPropertyValue(ProjectAssetsFile);
+
+            if (!File.Exists(assetsPath))
+            {
+                projectModel.AddProjectInformation(ProblemType.Error,
+                    string.Format(CultureInfo.CurrentCulture, Strings.Error_AssetsFileNotFound, projectPath));
+            }
+            else
+            {
+                var lockFileFormat = new LockFileFormat();
+                LockFile assetsFile = lockFileFormat.Read(assetsPath);
+
+                // Assets file validation
+                if (assetsFile.PackageSpec != null &&
+                    assetsFile.Targets != null &&
+                    assetsFile.Targets.Count != 0)
                 {
-                    var lockFileFormat = new LockFileFormat();
-                    var assetsFile = lockFileFormat.Read(assetsPath);
+                    // Get all the packages that are referenced in a project
+                    IEnumerable<FrameworkPackages> packages;
+                    try
+                    {
+                        packages = msBuild.GetResolvedVersions(project.FullPath, listPackageArgs.Frameworks, assetsFile, listPackageArgs.IncludeTransitive, includeProjects: listPackageArgs.ReportType == ReportType.Default);
+                    }
+                    catch (InvalidOperationException ex)
+                    {
+                        projectModel.AddProjectInformation(ProblemType.Error, ex.Message);
+                        return;
+                    }
+
 
-                    // Assets file validation
-                    if (assetsFile.PackageSpec != null &&
-                        assetsFile.Targets != null &&
-                        assetsFile.Targets.Count != 0)
+                    // If packages equals null, it means something wrong happened
```

Comment:
This seems like a remnant of some old code correct?

This method only handles 1 project. 
I'd simplify the comment if you still think it's necessary.

Reply:
Yes, it's remnants of old code.